spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -101,15 +107,40 @@ private void validateAndSetFilterParts() {
 
 	@Override
 	public Object execute(Object[] parameters) {
+		Class<?> returnedObjectType = getQueryMethod().getReturnedObjectType();
+		if (isSliceQuery() || isPageQuery()) {
```

Comment:
Note that `Slice` specifically doesn't require the total count. So, it is an unnecessary inefficiency to do this expensive count of total records, when it's not actually required.

Reply:
It doesn't require the total count, but it requires a boolean that indicates if the next slice exists. And the only way to know is to actually count the total number of records and compare it to (current offset + slice size). That's exactly what `Page` does to implement `Slice`.

Reply:
Don't you only need to query for {page size} + 1 records to know if there is another slice?